### PR TITLE
samples: cellular: modem_shell: Add Thingy:91 X Wi-Fi scanning support

### PR DESCRIPTION
**Pull request by tokangas**
Added devicetree overlay for building the sample for Thingy:91 X with Wi-Fi scanning support. Renamed the nRF700x Kconfig overlay, because the same overlay can be used with nRF7002 EK and Thingy:91 X.

### CODE REVIEW

File: samples/cellular/modem_shell/sample.yaml
Comment (trantanen) [inline samples/cellular/modem_shell/sample.yaml:296]: Should we add thingy91x compilation for some other basic configurations as well? I mean mainly the default configuration and maybe some other config that might be tight from memory usage perspective.

File: samples/cellular/modem_shell/thingy91x_wifi.overlay
Comment (trantanen) [inline samples/cellular/modem_shell/thingy91x_wifi.overlay:6]: Hmmm, could we actually have this always included (by being under boards folder) even if WiFi would be disabled by default?
I guess it shouldn't cause any power consumption penalty etc. if WiFi is not turned on by CONFIG_WIFI etc. Kconfig symbols?

File: samples/cellular/modem_shell/sample.yaml
Comment (tokangas) [reply samples/cellular/modem_shell/sample.yaml:296]: Yes, I'll add `thingy91x` to `sample.cellular.modem_shell.thingy91` and `sample.cellular.modem_shell.thingy91_modem_uart_trace` configurations.

As discussed offline, we could have a look if the `integration_platforms` lists are needed at all. Usually, both `integration_platforms` and `platforms_allow` lists contain the same platforms. I'm not sure `integration_platforms` is needed in this case, as the configuration should be built by default for all allowed platforms.

File: samples/cellular/modem_shell/thingy91x_wifi.overlay
Comment (tokangas) [reply samples/cellular/modem_shell/thingy91x_wifi.overlay:6]: I tried building the sample with and without the Wi-Fi include. There's a small (less than 200 bytes) difference in flash consumption. The increased flash consumption is not a problem as such, but I'm worried if the DTSI may affect current consumption even without the driver being enabled.

Comment (tokangas) [reply samples/cellular/modem_shell/thingy91x_wifi.overlay:6]: We decided to keep it like it's now, just in case.

**Review by peknis — DISMISSED** [review]

**Review by trantanen — APPROVED** [review]

**Review by moksanen — APPROVED** [review]
lgtm

**Review by jhirsi — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:429]:
```suggestion
  * Added:

    * Support for sending location data details into nRF Cloud with ``--cloud_details`` command-line option in the ``location`` command.
    * Support for Thingy:91 X Wi-Fi scanning.

  * Removed ESP8266 Wi-Fi DTC and Kconfig overlay files.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Comment by tokangas** [comment]
@nordicjm This is PR is ready to be merged, but it's not clear to me what kind of PRs can be merged and what not.

**Review by nordicjm — DISMISSED** [review]
This does not work with sysbuild, so cannot be accepted, unfortunately sysbuild does not support snippets and wifi configuration of which firmware is used is done at a sysbuild level, not an application level

**Review by trantanen — APPROVED** [review]

**Review by jhirsi — APPROVED** [review]

**Comment by tokangas** [comment]
Pending PR #13153.

**Comment by tokangas** [comment]
Rebased on sysbuild changes.

**Review by trantanen — APPROVED** [review]

File: samples/cellular/modem_shell/README.rst
Comment (nordicjm) [inline samples/cellular/modem_shell/README.rst:1089]: this isn't even correct, look at the sample.yaml file

**Review by nordicjm — CHANGES_REQUESTED** [review]
do not dismiss my nack, especially when it isn't even fixed

**Comment by tokangas** [comment]
> do not dismiss my nack, especially when it isn't even fixed

@nordicjm I tested the build command in the example with your PR before it was merged. It worked (and still works) without sysbuild and at that point sysbuild was not enabled by default. Now it seems that the default has been changed to sysbuild and the build command does not work anymore.

Communication about the whole sysbuild change has been bad. It has not been communicated what developers actually need to do. Requesting changes with comment "This does not work with sysbuild" was not very helpful either.

Because of this change, all build command examples in all samples with nRF700x are currently incorrect and need to be updated. This is probably not the only change needed. Is there going to be a treewide update of these or are individual developers just supposed to figure this out themselves?

File: samples/cellular/modem_shell/sample.yaml
Comment (tokangas) [inline samples/cellular/modem_shell/sample.yaml:160]: @nordicjm This option was added to three `modem_shell` configurations in #13153. I don't think these are needed and they do seem to cause a Kconfig warning. Please check.
```
warning: PM_EXTERNAL_FLASH_MCUBOOT_SECONDARY (defined at
/home/tok1/work/ncs/nrf/sysbuild/Kconfig.sysbuild:46) was assigned the value 'y' but got the value
'n'. Check these unsatisfied dependencies: BOOTLOADER_MCUBOOT (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_PM_EXTERNAL_FLASH_MCUBOOT_SECONDARY and/or
look up PM_EXTERNAL_FLASH_MCUBOOT_SECONDARY in the menuconfig/guiconfig interface. The Application
Development Primer, Setting Configuration Values, and Kconfig - Tips and Best Practices sections of
the manual might be helpful too.
```

**Comment by tokangas** [comment]
@nordicjm `modem_shell` documentation has now been updated.

BTW, I tried to re-request a review from you in the first place, but there was no "Re-request review" button visible in the PR. 
I guess this might have been because you had not submitted a single comment in the code when you finished the review.

File: samples/cellular/modem_shell/README.rst
Comment (peknis) [inline samples/cellular/modem_shell/README.rst:1084]:
```suggestion
To build the MoSh sample with Thingy:91 X Wi-Fi support, use the ``-DDTC_OVERLAY_FILE=thingy91x_wifi.overlay``, ``-DEXTRA_CONF_FILE=overlay-nrf700x-wifi-scan-only.conf``, ``-DSB_CONFIG_WIFI_NRF700X=y``, and ``-DSB_CONFIG_WIFI_NRF700X_SCAN_ONLY=y`` options.
```

**Review by peknis — APPROVED** [review]
Approving with a minor nit.

**Review by trantanen — APPROVED** [review]

File: samples/cellular/modem_shell/README.rst
Comment (tejlmand) [inline samples/cellular/modem_shell/README.rst:1163]: although the docs seems correct, then I wonder, why haven't this sample been adjust to have the following files instead:
- `prj_fota.conf` (merged content of `prj.conf` + `overlay-app_fota.conf`)
- `sysbuild_fota.conf`

then a `sysbuild_fota.conf` containing `SB_CONFIG_BOOTLOADER_MCUBOOT=y` can be made, and users could simply build the sample with fota support by doing:
```
$ west build -p -b *board_target* -- -DFILE_SUFFIX=fota
```

and similar goes for the other updated and complex examples in this PR.

**Review by tejlmand — APPROVED** [review]
lgtm.

An observation, although not directly related to the exact content of this PR.

File: samples/cellular/modem_shell/README.rst
Comment (trantanen) [reply samples/cellular/modem_shell/README.rst:1163]: > although the docs seems correct, then I wonder, why haven't this sample been adjust to have the following files instead:

Because we haven't known about `FILE_SUFFIX`. Seems it's not too old thing in Zephyr?
There are probably dozen(s) samples which could utilize this more so some info sharing would be nice on best practices in setting sample/app overlays (both Kconfig and DTC).

> * `prj_fota.conf` (merged content of `prj.conf` + `overlay-app_fota.conf`)

Copying a larger `prj.conf` with 50+ Kconfig options into all overlay config files doesn't sound too good. I guess there are no other possibilities to utilize `FILE_SUFFIX` than copying `prj.conf` into the desired `prj_<FILE_SUFFIX>.conf`?

Comment (tokangas) [reply samples/cellular/modem_shell/README.rst:1163]: Yes, otherwise that looks nice, but considering the size of `modem_shell` `prj.conf`, we probably don't want to duplicate it. We would lose the benefits of an overlay file, i.e. only having to include those options which are configured differently.

Comment (tokangas) [reply samples/cellular/modem_shell/README.rst:1163]: We decided that it's not feasible to use this mechanism for `prj.conf` and using it only for sysbuild options would probably not make the configuration clearer.

**Review by nordicjm — APPROVED** [review]

**Pull request merged**
